lxd/patches.go

```diff
@@ -1174,4 +1175,146 @@ func patchStorageZfsUnsetInvalidBlockSettingsV2(_ string, d *Daemon) error {
 	return nil
 }
 
+// patchStorageUnsetInvalidBlockSettings removes invalid block settings from LVM and Ceph RBD volumes.
+func patchStorageUnsetInvalidBlockSettings(_ string, d *Daemon) error {
+	s := d.State()
+
+	// Get all storage pool names.
+	pools, err := s.DB.Cluster.GetStoragePoolNames()
+	if err != nil {
+		// Skip the rest of the patch if no storage pools were found.
+		if api.StatusErrorCheck(err, http.StatusNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("Failed getting storage pool names: %w", err)
+	}
+
+	// Get the current leader.
+	var localConfig *node.Config
+	isLeader := false
+
+	err = d.db.Node.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.NodeTx) error {
+		localConfig, err = node.ConfigLoad(ctx, tx)
+		return err
+	})
+	if err != nil {
+		return err
+	}
+
+	if !d.serverClustered {
+		// If we're not clustered, we're the leader.
+		isLeader = true
+	} else {
+		leaderAddress, err := d.gateway.LeaderAddress()
+		if err != nil {
+			return nil
```

Comment:
this looks problematic, if there is an issue getting the leader address, we permanently skip applying the patch to this member's volumes?

Reply:
Good catch thanks! It should definitely return the actual error message. Must be a leftover from trying to escape the loop in earlier versions.

Reply:
That is fixed now.